rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -852,6 +878,36 @@ int SqliteStorage::read_db_schema_version()
   return schema_version;
 }
 
+uint64_t SqliteStorage::read_total_page_count_locked() const
+{
+  auto scope_exit_reset_statement = rcpputils::make_scope_exit(
+    [this]() {
+      page_count_statement_->reset();
+    });
+  auto page_count_query_result = page_count_statement_->execute_query<int>();
+  auto page_count_query_result_begin = page_count_query_result.begin();
+  if (page_count_query_result_begin == page_count_query_result.end()) {
+    throw SqliteException{"Error. PRAGMA page_count return no result."};
+  }
+  return std::get<0>(*page_count_query_result_begin);
+}
+
+uint64_t SqliteStorage::get_page_size() const
+{
+  if (!database_) {
+    return 0;  // Trying to call get_page_size when SqliteStorage::open was not called or db
+    // failed to open
+  }
+  SqliteStatement get_page_size_statement = database_->prepare_statement("PRAGMA page_size;");
```

Comment:
(making sure i understand correctly) i think that is intention not to have `get_page_size_statement_` member variable, instead using stack memory. because `get_page_size` is now called once during opening the sqlite database file, and not expected to be called many time after wards.

Reply:
@fujitatomoya Yes. your understanding is correct.  
And even if we would have a private member variable for the `get_page_size_statement` we would have to call database_->prepare_statement("PRAGMA page_size;"); and `get_page_size_statement->reset()` each time per `get_page_size` call due to the SQLite3 API limitation. Therefore it doesn;t make sense to cache it in the member variable.